src/man/sssd-ldap.5.xml

```diff
@@ -161,6 +161,45 @@
                     </listitem>
                 </varlistentry>
 
+                <varlistentry>
+                    <term>ldap_read_rootdse (string)</term>
+                    <listitem>
+                        <para>
+                            SSSD reads RootDSE to get information about LDAP and
+                            its capabilities. By default, this is done
+                            anonymously. However, this may not be permitted by
+                            the LDAP server. In such cases we can use this option
+                            to influence SSSD behavior.
+                        </para>
+                        <para>
+                            Allowed values are:
+                            <itemizedlist>
+                                <listitem>
+                                    <para>
+                                        anonymous
+                                    </para>
+                                </listitem>
+                                <listitem>
+                                    <para>
+                                        authenticated
+                                    </para>
+                                </listitem>
+                                <listitem>
+                                    <para>
+                                        never
```

Comment:
I know this is a typical value for startTLS/SSL but I don't think it makes much sense here. When set to never, it will never try to authenticate the request? So anonymous, is not forcing anonymous but as the default it will then try to authenticate?

both | default | all - one of these sounds more appropriate for the behavior
anonymous - should force anonymous
authenticated - should force authenticated

Reply:
No, no. With `never` It will never read RootDSE, but it will authenticate for other LDAP operations. This is not connected directly.

The purpose of this is that in some cases (e. g. LDAP proxy, security hardening) RootDSE is not accessible at all - authenticated or not. By setting it to `never` we stop flooding log with failures that are expected in such environment. In those cases we relay strictly on configuration file (some things like supported authentication or basedn can't be detected in this case and must be set correctly in the config file).

The `authenticated` works as you expect - it forces authentication before reading RootDSE

Only `anonymous` is a bit strange because it tries to read RootDSE after authentication - but only when it fails during anonymous attempt.

Reply:
Ah, let's add that clarification to the page. ;)